### ISSUE TITLE
Slow opening on  "/query $user"  with large history + custom timestamp

### ISSUE THREAD

**Issue by browseman** [hexchat/hexchat#1927]
Hello

I've spot a problem when opening a query windows / private chat with user that I have large chat history (1.3MB) in my case. This behavior popped up when I've changed the time stamp format from:
[%H:%M:%S] 

to:

[%Y%m%d-%H:%M:%S]

I'm using hexchat 2.10.1-1+deb8u1 on Debian Jessie 8.6 (last updates are applied) x86_64. I've tried to find some similar problem in the issue history, but without any success. If there is a need for any further information I will be very glad to provide further help.

**Comment by TingPing** [comment]
What does `[%Y%m%d-%H:%M:%S]` translate to in your locale? Does it have non-ascii characters?

**Comment by browseman** [comment]
Here is an example from yesterday...
[20170105-23:20:20]

**Comment by TingPing** [comment]
Ah its just long. This was probably fixed by f7bea16e37342d752568723d92ce3eca9e4a972a, simply update to the current release.

**Comment by browseman** [comment]
Guess I will have to wait for the Debian package maintainer to release the new version, or compile it by my self. 
Thanks for your help TingPing.

**Comment by TingPing** [comment]
You will ofc never get a new version unless the next major release, so you'll just have to backport it yourself.

**Comment by browseman** [comment]
Yes, I've just find and install a newer version from jessie-backports "v2.12.0" using the following command:
aptitude -t jessie-backports install hexchat.

The problem seems to be resolved, although the opening time of query window for the same user is about 2-3sec. This is ok for me, but I guess if the scrollback continue to grow, the opening time will continue to grow. Is this normal?

**Comment by TingPing** [comment]
It should be basically instant.
